srv/scripts/export_ads_annotations.js

```diff
@@ -0,0 +1,83 @@
+const MongoClient = require('mongodb').MongoClient;
+const stringify = require('csv-stringify')
+const fs = require('fs');
+
+const MONGODB_URL = 'mongodb://localhost:27017';
+const MONGODB_DB_NAME = 'facebook_ads';
+
+const client = new MongoClient(MONGODB_URL, { useNewUrlParser: true });
+
+const DELIMITER = ';';
+const COLUMNS = ['facebook_ad_id', 'crowdsourcing_annotation', 'reviewed_annotation'];
+
+const exportToCSV = (annotations, reviews) => {
+	return new Promise((resolve, reject) => {
+		const data = []
+
+		const stringifier = stringify({
+			delimiter: DELIMITER,
+			header: true,
+			columns: COLUMNS
+		});
+
+		stringifier.on('readable', () => {
+			let row;
+			while(row = stringifier.read()){
+				data.push(row)
+			}
+		});
+
+		stringifier.on('error', (err) => {
+			reject(err.message);
+		});
+
+		annotations.forEach(annotation => {
+			const review = reviews.find(r => r.adId === annotation.adId);
```

Comment:
The search for the annotation requires `reviews.length` comparisons. This is done `annotations.length` times (inside a `forEach`). The time complexity of this script is then `reviews.length` * `annotations.length`. This could be close to `annotations.length`, using a hashmap of reviews. However, since these arrays are in the thousands, the execution should be quick enough.

Reply:
Yep, indeed :)